C-Sharp.md

```diff
@@ -1,6 +1,232 @@
 #The C# Programming Language
 
+1. ### Basics
+  * Setup
+  * Your first C# program
+  * Types and variables
+  * Flow control statements
+  * Strings
+  * Classes, Objects, Interface and methods
+  * Fields and properties
+  * Scope and accessibility modifiers
+  * Handling exceptions
 
-* Introduction
-* Setup
-* Language features 
+2. ### Intermediate
+  * Generics
+  * Events, delegates and lambda expressions
+  * Collection framework
+  * LINQ
+
+3. ### Advanced
+  * Asynchronous Programming (Async and Await)  
+  * Task Parallel Library
+
+5. ### What's New in C# 6
+  * Null-Conditional operator
+  * Auto-Property initializers
+  * Nameof expressions
+  * Expression bodied functions and properties
+  * Other features
+
+4. ### Object-Oriented principles (OOP)
+   * Encapsulation
+   * Abstraction
+   * Inheritance
+   * Polymorphism
+
+6. ### Solid principles
+  * Single responsibility principle
+  * Open closed principle
+  * Liskov substitution principle
+  * Interface segregation principle
+  * Dependency inversion principle
+
+7. ### C# Best practices, Design patterns & Test driven development (TDD)
+
```

Comment:
For now, let's merge this list as is. But later we would have to move this list to the [.NET](https://github.com/FreeCodeCamp/FreeCodeCamp/wiki/Dotnet) homepage and link each topic to their corresponding Wiki articles.